lib/LoginStrategy.ts

```diff
@@ -239,4 +254,78 @@ export class LoginStrategy extends BaseStrategy {
       return acc
     }, [] as Claim[])
   }
-}
+
+  /**
+   * @description checks if the claimer is authenticated into the DID Document
+   *
+   * @param token
+   * @param payload
+   *
+   * @returns {string} issuer DID or null
+   */
+  async verifyClaim(token: string, payload: ITokenPayload): Promise<string> {
+    const didDocument = await this.cacheServerClient?.getDidDocument(payload.iss) ?? await this.didResolver.read(payload.iss)
+
+    if (await this.isAuthorized(token, didDocument))
+      return payload.iss
+    return null 
+  }
+  
+  private isAuthorized = async (claimedToken: string,didDocument: IDIDDocument) : Promise<boolean> => {
+    //read publickey field in DID document
+    const didPubKey = didDocument.publicKey
+    if (didPubKey.length === 0)
+      return false
+
+    const keys = new Keys({privateKey: this.privateKey})
+    const jwtSigner =  new JWT(keys)
+
+    //get all authenticated public keys
+    const authenticatedPubkeys = didDocument.publicKey.filter(pubkey => {
+      return this.isAuthenticated(pubkey, didDocument.authentication)
+    })
```

Comment:
A bit of a minor point, but I think it's clearer if the wording used is "authentication" public keys. This is because the keys are used for authentication, the key are not themselves authenticated.
```suggestion
    //get all authentication public keys
    const authenticationPubkeys = didDocument.publicKey.filter(pubkey => {
      return this.isAuthenticationKey(pubkey, didDocument.authentication)
    })
```